datajunction-server/datajunction_server/sql/dag.py

```diff
@@ -140,3 +148,55 @@ def get_shared_dimensions(
         [y for x in common.values() for y in x],
         key=lambda x: (x.name, x.path),
     )
+
+
+def get_nodes_with_dimension(
+    session: Session,
+    dimension_node: Node,
+    node_types: Optional[List[NodeType]] = None,
+) -> List[NodeRevision]:
+    """
+    Find all nodes that can be joined to a given dimension
+    """
+    to_process = [dimension_node]
+    processed: Set[str] = set()
+    final_set: Set[NodeRevision] = set()
+    while to_process:
+        current_node = to_process.pop()
+        processed.add(current_node.name)
+
+        # Dimension nodes are used to expand the searchable graph by finding
+        # the next layer of nodes that are linked to this dimension
+        if current_node.type == NodeType.DIMENSION:
+            statement = (
+                select(NodeRevision)
```

Comment:
Since this selects `NodeRevision`, does that mean it will include older revisions? I think we should only be considering the latest (or currently active?) node revision for each node name.

Reply:
Yep agreed! Just modified this to only take into account the latest revision